Makefile

```diff
@@ -8,9 +8,25 @@ PROJECT_URL := "https://github.com/scootdev/scoot"
 default:
 	go build ./...
 
+dependencies: 
+	# get all dependencies needed to run scoot / check if there have
+	# been updates since they were first installed
+	go get -t -u github.com/scootdev/scoot/...
+
+	# mockgen is only referenced for code gen, not imported directly
+	go get github.com/golang/mock/mockgen
+
+generate: 
```

Comment:
actually on second thought, dependencies takes a while to run because it has to check if any of the packages changed over the network too, and I often want to generate code while not connected

so for now just leaving it off, I added fullbuild target which will check dependencies, generate code, and run all tests, this is basically our checkin build, grab latest dependencies, generate code with them, then run tests.